### PR TITLE
Codelab module and codelab refactors

### PR DESCRIPTION
**Pull request by TristonianJones**
Update the codelab to be its own go-module and refactor the codelab solution to use
the latest features such as `cel.Variable` and `cel.Function` which both simplify the code
and perform more of the boilerplate tasks in the core CEL library rather than in user-
provided extensions.

Will update the public codelab to match post-submission as part of fixes for #553

### CODE REVIEW

**Review by l46kok — DISMISSED** [review]
Could we also [update the examples](https://github.com/google/cel-go/tree/master/examples) with this new syntax? Either in this or in a separate PR

**Comment by TristonianJones** [comment]
@l46kok good catch on updating the examples as well. Done.

File: examples/README.md [hunk 1]
```diff
@@ -6,14 +6,13 @@ Evaluate expression `"Hello world! I'm " + name + "."` with `CEL` passed as
...
-    "github.com/google/cel-go/cel"
-    "github.com/google/cel-go/checker/decls"
-)
-
-    d := cel.Declarations(decls.NewVar("name", decls.String))
-    env, err := cel.NewEnv(d)
+import "github.com/google/cel-go/cel"
 
+    env, err := cel.NewEnv(cel.Variable("name", cel.StringType))
+    // Check iss for compilation errors.
```
Comment (l46kok) [inline examples/README.md]: Nit: 'err' and looks like an environment creation error

File: examples/README.md [hunk 2]
```diff
@@ -40,35 +39,31 @@ Evaluate expression `i.greet(you)` with:
...
+                cel.StringType,
+                cel.BinaryBinding(func (lhs, rhs ref.Val) ref.Val {
+                    return types.String(
+                        fmt.Sprintf("Hello %s! Nice to meet you, I'm %s.\n", rhs, lhs))
+                    },
+                ),
+            ),
+        ),
+    )
+    // Create env and compile
```
Comment (l46kok) [inline examples/README.md]: Nit: Looks like we can just remove this comment or update

Comment (TristonianJones) [reply examples/README.md]: Done.

File: examples/README.md (hunk 1 above)
Comment (TristonianJones) [reply examples/README.md]: Done.

**Review by l46kok — APPROVED** [review]

**Pull request merged**